tracer/contrib/tracegrpc/grpc.go

```diff
@@ -68,51 +68,80 @@ func serverSpan(t *tracer.Tracer, ctx context.Context, method, service string) *
 	span.SetMeta("gprc.method", method)
 	span.Type = "go"
 
-	traceID, parentID := getIDs(ctx)
+	traceID, parentID, samplingPriority, hasSamplingPriority := getCtxMeta(ctx)
 	if traceID != 0 && parentID != 0 {
 		span.TraceID = traceID
+		t.Sample(span) // depends on trace ID so needs to be updated to maximize the chances we get complete traces
 		span.ParentID = parentID
+		if hasSamplingPriority {
+			span.SetSamplingPriority(samplingPriority)
+		}
 	}
 
 	return span
 }
 
-// setIDs will set the trace ids on the context{
-func setIDs(span *tracer.Span, ctx context.Context) context.Context {
+// setCtxMeta will set the trace ids and the sampling priority on the context.
+func setCtxMeta(span *tracer.Span, ctx context.Context) context.Context {
 	if span == nil || span.TraceID == 0 {
 		return ctx
 	}
 
-	md := metadata.New(map[string]string{
-		traceIDKey:  fmt.Sprint(span.TraceID),
-		parentIDKey: fmt.Sprint(span.ParentID),
-	})
+	var md metadata.MD
+
+	if span.HasSamplingPriority() {
```

Comment:
This is where I got it wrong initially, one really has to test wether the data is here or not, else all children could have a sampling priority set to 0 by default. And so, never make their way to the server. This is tricky because they are silently dropped so you never see them and you can't find broken traces as it's the child part that disappears.